### PR TITLE
Feature step6 integration

### PR DESCRIPTION
**Pull request by kmilo9999**
- Added file analysis_db/B05_define_angle.py for step 6
- Cleaned code
- Formatted file

### CODE REVIEW

File: src/icesat2_tracks/analysis_db/B05_define_angle.py
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: In #51 there is a file named `B04_angle.py`. Should the file at hand  be renamed in that format (no extra `define_`)? Should the other file be renamed to fit the format in this file? @kmilo9999 @hollandjg @mochell

Comment (mochell) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: Im not sure what you mean. you want to call it B04_define_angle.py ?

Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: Yes, just to have a uniform naming convention and make things a bit more intuitive and predictable.

**Review by cpaniaguam — DISMISSED** [review]
B05_define_angle needs to be cleaned up. I'll do this when the file is added to the cli.

File: src/icesat2_tracks/analysis_db/B05_define_angle.py
Comment (hollandjg) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: Is there a more descriptive name we could use? Perhaps an extra verb in the name for B04 so you can see from its name what it's doing?

Comment (hollandjg) [inline src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: I think you've duplicated this file, and not deleted the old one, @kmilo9999! I can't see any changed you made in this file.

**Review by hollandjg — CHANGES_REQUESTED** [review]
Please could you delete the old version of the file so we can see the diff in the GitHub files view? Thanks!

File: src/icesat2_tracks/analysis_db/B05_define_angle.py
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: I deleted the old file. Let me know if you still cant see the changes in this file

File: src/icesat2_tracks/analysis_db/B05_define_angle.py [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+    weights = weights - weights.min()
+    return weights
+
+
+def weighted_means(data, weights, x_angle, color="k"):
+    """
+    weights should have nans when there is no data
+    data should have zeros where there is no data
+    """
+    from scipy.ndimage import label
```
Comment (hollandjg) [inline src/icesat2_tracks/analysis_db/B05_define_angle.py:77]: Depending on the results of the other discussion, I think we might want to move this import to the top of the file.

File: src/icesat2_tracks/analysis_db/B05_define_angle.py [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+    mconfig,
+    color_schemes,
+    plt,
+    font_for_print    
+)
+
+import icesat2_tracks.ICEsat2_SI_tools.io as io
+import icesat2_tracks.ICEsat2_SI_tools.spectral_estimates as spec
+
+from numba import jit
```
Comment (hollandjg) [inline src/icesat2_tracks/analysis_db/B05_define_angle.py:18]: Do we need this line? I don't think we're using the import anywhere.

**Review by hollandjg — COMMENTED** [review]
I've a couple of questions around the edges – nothing major. It looks like just moving the file b05 and then reformatting it.

File: src/icesat2_tracks/analysis_db/B05_define_angle.py (hunk 1 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py]: Done!

File: src/icesat2_tracks/analysis_db/B05_define_angle.py (hunk 2 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py]: I removed this import and the code still runs locally. Let's see how it behaves on  the CI

**Review by hollandjg — APPROVED** [review]
Looks good to me!

File: src/icesat2_tracks/analysis_db/B05_define_angle.py
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: #97

[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: #98

Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B05_define_angle.py:1]: Camilo needs to use `git move`

**Pull request merged**